src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -65,12 +68,14 @@ private sealed class CriticalHelper
             private ArgBuilder? _collectionContractArg;
 
             [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+            [RequiresDynamicCode(DataContract.SerializerAOTWarning)]
             private static XmlFormatClassReaderDelegate CreateReflectionXmlClassReader(ClassDataContract classContract)
             {
                 return new ReflectionXmlClassReader(classContract).ReflectionReadClass;
             }
 
             [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+            [RequiresDynamicCode(DataContract.SerializerAOTWarning)]
```

Comment:
Is this one RDC even with the knowledge that the `DataContractSerializer.Option == SerializationOption.ReflectionOnly` codepath will always be taken when AOT compiling?

`DataContractSerializer.Option` basically controls whether we're going to take the ref.emit code paths.

Reply:
Both implementations are flagged by the analyzer as annotated with RDC, so the option seems to not affect the end result. I tried to go depper on the ReflectionOnly call and at several points it calls multiple methods annotated with RDC